sale_coupon_multi_use/models/sale_coupon.py

```diff
@@ -57,9 +59,22 @@ def _get_related_sale_order_line(self, sale_order):
 
     def _prepare_consumption_line(self, sale_order_line):
         self.ensure_one()
+        program = self.program_id
+        from_currency = sale_order_line.order_id.pricelist_id.currency_id
```

Comment:
```suggestion
        from_currency = sale_order_line.order_id.currency_id
```

Reply:
Can't find source, but I remember it was said that explicit full path to record is better performance wise instead of using related field, because you don't need to initiate related field logic.

Reply:
my point here that currency on sale order may differ from price list currency because of customer settings or manual change etc.

Reply:
How can it differ? Its related field that points to pricelist currency:

```python
    currency_id = fields.Many2one("res.currency", related='pricelist_id.currency_id', string="Currency", readonly=True, required=True)
```

Reply:
oh, sorry than I missed some, I thought it was optional